src/coreclr/src/pal/inc/pal_mstypes.h

```diff
@@ -584,6 +584,10 @@ typedef LONG_PTR LPARAM;
 #define _PTRDIFF_T
 #endif
 
+#if defined(__PALSUITE_H__) && !defined(__cplusplus)
```

Comment:
I would put this typedef into the palsuite.h instead and keep this file untouched.

Reply:
Moved the condition to palsuite.h